libs/datasets/timeseries.py

```diff
@@ -1559,12 +1587,14 @@ def _with_common_index(
     datasets_wide: Mapping[MultiRegionDataset, pd.DataFrame]
 ) -> Mapping[MultiRegionDataset, pd.DataFrame]:
     dataset_wide_first = more_itertools.first(datasets_wide.values())
-    assert dataset_wide_first.index.names == [CommonFields.LOCATION_ID]
+    assert dataset_wide_first.index.names == [CommonFields.LOCATION_ID, PdFields.DISTRIBUTION]
     assert dataset_wide_first.columns.names == [PdFields.VARIABLE]
-    common_index = pd.Index(
-        np.unique(np.concatenate([ds_w.index for ds_w in datasets_wide.values()])),
-        name=dataset_wide_first.index.name,
-    )
+    common_index = None
```

Comment:
nit: Colud set this to `dateset_wide_first.index` and then remove the `is None` case below?

Reply:
If I changed it I'd also need to skip the first when iterating through `values()` so decided to stick with the branch instead of adding more abstractions.

Reply:
Out of curiosity, wouldn't `union()` gracefully ignore duplicates?  Definitely no problem with the existing code though.

Reply:
I'd expect so but it seems clearer to write what I want to happen.